butane_core/src/migrations/mod.rs

```diff
@@ -326,3 +326,37 @@ impl crate::internal::DataObjectInternal for ButaneMigration {
         Ok(()) // no-op
     }
 }
+
+/// Migrate forward.
+pub fn migrate<M: Migration>(
+    connection: &mut impl crate::db::BackendConnection,
+    migrations: &impl Migrations<M = M>,
+) -> Result<()> {
+    let to_apply = migrations.unapplied_migrations(connection)?;
+    for migration in &to_apply {
+        crate::info!("Applying migration {}", migration.name());
+        migration.apply(connection)?;
+    }
+    Ok(())
+}
+
+/// Rollback all applied migrations.
+pub fn rollback<M: Migration>(
```

Comment:
perhaps this should be called `downgrade` to be consistent and avoid conflicting terminology with Transaction.rollback .

However the CLI uses `rollback`.  perhaps that should also be changed.

That said, "downgrade" doesnt sound right to me.  It feels more like a large version thing, not a single migration thing.

And the opposite of downgrade should be upgrade, not "migrate".  Perhaps "unmigrate" is clearer?  Or "revert"? or "unapply"?

As a refresher, django uses `migrate app.old_migration_name` to remove migrations.

Reply:
Good question. I think `unmigrate` seems clearest to me, or maybe `unapply_all`.

Reply:
After this PR, I'll update the CLI to use whatever terminology we adopt here, deprecating "rollback".

Reply:
re `apply_all` or `apply_unapplied`, I'd prefer we stick with single words, and "migrate" is very common